apps/site/lib/site_web/templates/fare/_nearby_locations.html.eex

```diff
@@ -1,16 +1,55 @@
-<div class="fare-sales-locations">
-  <h3>Nearby MBTA fare retail sales locations</h3>
-  <div class="c-location-cards c-location-cards--background-gray small-set grouped">
-    <%= for {%Fares.RetailLocations.Location{name: name, address: address} = location, distance} <- @fare_sales_locations do %>
-      <%= link to: GoogleMaps.direction_map_url(@search_position, location), class: "c-location-card c-location-card--flex", target: "_blank" do %>
-        <div class="c-location-card__name">
-          <%= name %>
+<div class="c-location-cards">
+  <%= for {location, distance} <- @locations do %>
+    <%
+      name = Map.get(location, :name)
+      address = Map.get(location, :address)
+      retail_fvm = Map.get(location, :retail_fvm)
+      stop_id = Map.get(location, :stop_id)
+    %>
+    <div class="c-location-card c-location-card--rsl">
+      <div class="c-flex-space-between">
+        <div class="c-location-card__place">
+          <%= if stop_id do %>
+            <%= link name, to: stop_path(@conn, :show, stop_id) %>
+          <% else %>
+            <%= name %>
+          <% end %>
         </div>
         <div class="c-location-card__distance"><%= round_distance(distance) %></div>
-        <div class="c-location-card__description">
+      </div>
+      <%= if address do %>
+        <div class="c-location-card__location">
           <%= address %>
         </div>
       <% end %>
-    <% end %>
-  </div>
+      <%= if retail_fvm do %>
+        <div class="c-location-card__type">
+          <%= retail_fvm %>
+        </div>
+      <% end %>
+
+      <div class="dropdown show c-location-card__directions">
+        <a href="#" role="button" id="get-directions-dropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+          Get Directions
+          <span class="c-caret"></span>
+        </a>
+        <ul class="dropdown-menu" aria-labelledby="get-directions-dropdown">
+          <li>
+          <% destination_address = if !is_nil(address), do: address, else: name %>
+            <%= link "View on MBTA Trip Planner", to: trip_plan_path(@conn, :index, %{
+              from: @address,
+              from_latitude: Float.to_string(@search_position.latitude, decimals: 4),
+              from_longitude: Float.to_string(@search_position.longitude, decimals: 4),
+              to: destination_address,
+              to_latitude: Float.to_string(location.latitude, decimals: 4),
+              to_longitude: Float.to_string(location.longitude, decimals: 4),
+            }), target: "_blank", class: "dropdown-item" %>
```

Comment:
Although clicking on this correctly populates the 'to' address, the reason why we implemented the `/to` feature in the trip planner was to not do it in the frontend (context [here](https://app.asana.com/0/555089885850811/1199926921659622)). Also this won't put the B pin in the map.
I believe it should be:
```elixir
<%= link "View on MBTA Trip Planner", to: trip_plan_path(@conn, :to, "#{destination_address}"), target: "_blank", class: "dropdown-item" %>
```

Reply:
The `/to` logic was actually broken for the PSLs, and I didn't notice until trying to implement for RSLs.  [Here's a link to the PSLs near me](https://www.mbta.com/fare-transformation/proposed-sales-locations?latitude=42.340575699999995&longitude=-71.1567228#address-search-input) in West Boston.  Clicking for MBTA directions directs me to https://www.mbta.com/trip-planner/to/South%20Street, which is a location out in Fitchburg

Reply:
mmm I don't think it's necessarily broken, it just takes a guess. You might have just found an edge case in which "South Street" is just too generic by name (other PSL's near you seem to work fine?)

Reply:
Oh true, I shouldn't say broken.  But there are many instances of wrong guesses.  Here is what I've found so far:

Back Bay:  [Station](https://www.mbta.com/stops/place-bbsta) vs [Trip Planner Directions](https://www.mbta.com/trip-planner/to/Back%20Bay)
Assembly:  [Station](https://www.mbta.com/stops/place-astao) vs [Trip Planner Directions](https://www.mbta.com/trip-planner/to/Assembly)
Chinatown: [Station](https://www.mbta.com/stops/place-chncl) vs [Trip Planner Directions](https://www.mbta.com/trip-planner/to/Chinatown)
State:  [Station](https://www.mbta.com/stops/place-state) vs [Trip Planner Directions](https://www.mbta.com/trip-planner/to/State)

There are so many generic station names that really throw off the map predictions of "which location are we talking about?"

Reply:
Well, unfortunately it doesn't seem like we have the specific address name of a PSL available, so the trip planner does as best as it can by name 😅 

But one thing I just thought of is that we could try to make it more accurate by concatenating the municipality, as in
`.../trip-planner/to/{stop name}, {municipality}`  ?  Just a thought but it might help.

Reply:
Mmmmm the trickiness of this problem is resonating with me now.  Thanks for your details, these are helping a lot.

Using municipality is a good lead.  Unfortunately, the Back Bay and Chinatown locations still pose a problem, since the wrong guess and the actual location are in the same municipality 😞 

But I tried something else with the url and might have a good lead.  I found a way to use lat and lng values in the url, and the pin shows up on the map!  One remaining hurdle is that it populates the trip planner "to" input box with a weird value, but I had a geographer friend explain what that value means, and I might have a way to convert it to the stop name!  Let's see...